rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/backend/analysis/AnalysisRpcService.java

```diff
@@ -104,4 +104,40 @@ public interface AnalysisRpcService {
    */
   @JsonNotification
   void didChangeAutomaticAnalysisSetting(DidChangeAutomaticAnalysisSettingParams params);
+
+  /**
+   * Analyze all files in the project. User file exclusions and .gitignore will be respected.
+   * @param params configuration scope ID, flag to report only hotspots
+   * Issues will be reported to the client via
+   * {@link SonarLintRpcClient#raiseIssues(RaiseIssuesParams)} and {@link SonarLintRpcClient#raiseHotspots(RaiseHotspotsParams)}
+   */
+  @JsonNotification
+  void analyzeFullProject(AnalyzeFullProjectParams params);
+
+  /**
+   * Analyze all files in the provided list. User file exclusions and .gitignore will be respected.
+   * @param params configuration scope ID, list of files to analyse
+   * Issues will be reported to the client via
+   * {@link SonarLintRpcClient#raiseIssues(RaiseIssuesParams)} and {@link SonarLintRpcClient#raiseHotspots(RaiseHotspotsParams)}
+   */
+  @JsonNotification
+  void analyzeFileList(AnalyzeFileListParams params);
```

Comment:
In SLI, there is one use case for analyzing the active file (the green triangle), but this method may be able to achieve it.
For this use case, I think exclusions are not respected, to be verified. So it might deserve a dedicated method. Or we just stop having this behavior

Reply:
I don't think we check exclusions for the forced analysis.